frontend/src/scenes/data-warehouse/editor/sidebar/QueryInfo.tsx

```diff
@@ -151,6 +155,34 @@ export function QueryInfo({ codeEditorKey }: QueryInfoProps): JSX.Element {
                                         loading={updatingDataWarehouseSavedQuery}
                                         options={OPTIONS}
                                     />
+                                    <LemonButton
+                                        type="secondary"
+                                        size="small"
+                                        tooltip="Revert materialized view to view"
+                                        disabledReason={
+                                            savedQuery?.status === 'Running' &&
+                                            'Cannot revert while materialization is running'
+                                        }
```

Comment:
It'd be nice to be able to revert this when its running - especially if the materialization gets stuck somewhere

Reply:
I think the cancellation should cover us there. You could click cancel and then revert in quick succession. Another way of doing this is leaving this active all the time and making it check for running tasks (ie: do the cancel workflow as part of this)... I think either option is reasonable, wdyt?